src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransferStore.cs

```diff
@@ -19,6 +18,40 @@
 
 namespace Stratis.FederatedPeg.Features.FederationGateway.TargetChain
 {
+    public class StatusChangeTracker:Dictionary<ICrossChainTransfer, CrossChainTransferStatus?>
+    {
+        /// <summary>
+        /// Records changes to transfers for the purpose of synchronizing the transient lookups after the DB commit.
+        /// </summary>
+        /// <param name="transfer">The cross-chain transfer to update.</param>
+        /// <param name="status">The new status.</param>
+        /// <param name="blockHash">The block hash of the partialTranction.</param>
+        /// <param name="blockHeight">The block height of the partialTransaction.</param>
+        public void SetTransferStatus(ICrossChainTransfer transfer, CrossChainTransferStatus? status = null, uint256 blockHash = null, int blockHeight = 0)
+        {
+            if (status != null)
+            {
+                // If setting the status then record the previous status.
+                this[transfer] = transfer.Status;
+                transfer.SetStatus((CrossChainTransferStatus)status, blockHash, blockHeight);
```

Comment:
`SetStatus()`, calls `IsValid()` - `IsValid()` doesn't do anything with the `blockHeight`.  Is this parameter required ?

Reply:
It is used in the method:

```
        public void SetStatus(CrossChainTransferStatus status, uint256 blockHash = null, int blockHeight = 0)
        {
            this.status = status;

            if (this.status == CrossChainTransferStatus.SeenInBlock)
            {
                this.blockHash = blockHash;
                this.blockHeight = blockHeight;
            }

            Guard.Assert(IsValid());
        }
```

Reply:
Ar, okay, I missed `this.blockHeight`, thanks.